scrapy/extensions/checksettings.py

```diff
@@ -0,0 +1,54 @@
+"""
+Check settings extension
+
+See documentation in docs/topics/extensions.rst
+"""
+import logging
+import pprint
+import operator
+
+from fuzzywuzzy.fuzz import ratio
+from scrapy import signals
+from scrapy.exceptions import NotConfigured
+
+logger = logging.getLogger(__name__)
+
+
+class CheckSettings:
+    def __init__(self, crawler):
+        self.settings = crawler.settings
+        self.check_settings_min = crawler.settings.getint("CHECK_SETTINGS_MIN")
+        self.not_used_settings = [s for s in self.settings
```

Comment:
Wouldn’t it be better to delay this as much as possible (i.e. on spider close)? That way we minimize the chance of false positives.

Reply:
You're right, will change it.

Reply:
I think that probably in cases of spider with thousands of links to follow the user will like to have the list of not used settings in the beginning. What do you think?

Reply:
You have a point. But I’m worried that settings used later will often get misreported, which could be confusing. I’m unsure; go with your gut, and once this goes out of WIP let’s see what other people think.